aten/src/ATen/core/function_schema.h

```diff
@@ -550,15 +550,21 @@ inline std::ostream& operator<<(std::ostream& out, const Argument& arg) {
   bool is_opt = type->kind() == OptionalType::Kind;
   auto unopt_type = is_opt ? type->castRaw<OptionalType>()->getElementType() : type;
 
-  if (unopt_type->kind() == ListType::Kind && arg.N()) {
+  if (unopt_type->kind() == ListType::Kind) {
     // sized lists get size N from arg, not type
     auto list = unopt_type->cast<c10::ListType>();
-    out << list->getElementType()->str() << "[" << *arg.N() << "]";
+    out << list->getElementType()->str();
+    if (arg.alias_info() && !arg.alias_info()->containedTypes().empty()){
+      out << arg.alias_info()->containedTypes()[0] << "[]";
+    } else if (arg.N()) {
+      out << "[" << *arg.N() << "]";
+    }
   } else {
     out << unopt_type->str();
   }
 
-  if (arg.alias_info()) {
+  // print alias info if it has beforeSets.
+  if (arg.alias_info() && !arg.alias_info()->beforeSets().empty()) {
```

Comment:
Should we have an API in AliasInfo indicating if the alias info worth printing? We can check if beforeSets is empty inside that API

Reply:
Is it possible to have an alias info with nothing in the BeforeSet? Maybe we should just ban that state altogether (assert in the `AliasInfo` constructor that the before set is non-empty)

Reply:
> Is it possible to have an alias info with nothing in the BeforeSet?

Yeah I don't see any use case. Let me add an assert and see what's happen in CI

Reply:
Oh it's an implicit constructor though. I don't see problem of not printing alias info that has an empty before sets, I can add a unit test to assert that it's not printing

Reply:
Ah ok! Seems fine then.